RtspServer/Classes/SourceMedia.cs

```diff
@@ -239,15 +238,15 @@ public SourceMedia(string name, Uri source, NetworkCredential sourceCredential,
 
         internal void OnFrameDecoded(System.Drawing.Image decoded)
         {
-            if (DecodeFrames && decoded != null && FrameDecoded != null)
+            if (DecodeFrames && decoded != null)
             {
                 FrameDecoded.Invoke(this, decoded);
             }
         }
 
         internal void OnFrameDecoded(byte[] decoded)
         {
-            if (DecodeFrames && decoded != null && DataDecoded != null)
+            if (DecodeFrames && decoded != null)
```

Comment:
If we are going to remove the null check then lets put back the condition on the Invoke?, its up to you.

Reply:
.?Invoke is similar to

```
var @delegate = delegate;
if (@delegate != null) @delegate.Invoke()
```
So it both race-free and checks for null.

Reply:
Which do you prefer? My feeling is that since the delegate is going to be checked for null either way that we should use which ever approach is more readable, I think both are also equally readable.

Reply:
If we are going to check for null explicitly then we don't need to use `?` to optionally invoke as we know the value is null or not, so lets try to base it on that.

Reply:
I think we missed the check for null here, it was here previously but we changed to use ?.Invoke, I think I changed that back on accident so we either need a null check for ``&& DataDecoded != null`` or we need to use the ```?. ```